frontend/public/module/k8s/index.ts

```diff
@@ -337,6 +338,35 @@ export type PodKind = {
 } & K8sResourceCommon &
   PodTemplate;
 
+export type DeploymentKind = {
```

Comment:
@spadgett Shouldn't we define these new types inside `console-shared` because it will help in reducing the dependency of other packages on console-internal and `DeploymentKind` is already being used in dev console/console-shared and might be usable in `knative-plugin`.
@cvogt ^^

Reply:
`public` is fine for now because that's where all the other types are. `public` will _eventually_ be addressed in the _future_